java/core/src/main/java/co/worklytics/psoxy/gateway/impl/oauth/RefreshTokenTokenRequestBuilder.java

```diff
@@ -35,8 +35,21 @@
     public enum ConfigProperty implements ConfigService.ConfigProperty {
         REFRESH_TOKEN, //NOTE: you should configure this as a secret in Secret Manager
         CLIENT_SECRET, //NOTE: you should configure this as a secret in Secret Manager
+        USE_SHARED_TOKEN,
     }
 
+    @Override
+    public boolean useSharedToken() {
+        Optional<String> useSharedTokenConfig = config.getConfigPropertyAsOptional(ConfigProperty.USE_SHARED_TOKEN);
```

Comment:
Adding support for shared token with `refresh_token` grant_type; used in jira cloud.